openapi_python_client/parser/properties/__init__.py

```diff
@@ -352,7 +351,13 @@ def build_union_property(
     *, data: oai.Schema, name: str, required: bool, schemas: Schemas, parent_name: str, config: Config
 ) -> Tuple[Union[UnionProperty, PropertyError], Schemas]:
     sub_properties: List[Property] = []
-    for i, sub_prop_data in enumerate(chain(data.anyOf, data.oneOf)):
+
+    sub_data: List[Union[oai.Schema, oai.Reference]] = []
+    for _data in [data.anyOf, data.oneOf]:
+        if isinstance(_data, Iterable):
+            sub_data.extend(_data)
+
+    for i, sub_prop_data in enumerate(sub_data):
```

Comment:
```suggestion
    for i, sub_prop_data in enumerate(chain(data.anyOf or [], data.oneOf or [])):
```
Should be quite a bit faster since checking truthiness is less expensive than isinstance and we don't need to allocate / extend an extra list.

Although, is there a good reason for not defaulting to empty list like before? We aren't checking for explicit `None` anyway, right? So semantically they should be the same.

Reply:
Agreed, resolved in https://github.com/triaxtec/openapi-python-client/pull/426/commits/3570a59bca36ac38738e0e30a3b615b138a75348

I thinks its better to not default to an empty list on the `oai.Schema`; it could lead to  an unexpected and tricky behaviour since multiple instances of it will share the same list reference.

I did not see any explicit `None` check, all tests are green, I thinks that we are fine here

Reply:
I thought that Pydantic did something special to work around mutable defaults. If that's not the case then we could do

```python
from pydantic import Field

anyOf: List[Union[Reference, "Schema"]] = Field(default_factory=list)
```

Reply:
Yes indeed, in the actual version with have:
- allOf defaulting to `None`
- anyOf, oneOf defaulting to `List.Empty`

I think it should be normalized, either make them all defaulting to `None` or `List.Empty`.  What do you think?

Reply:
I think empty list makes more sense for us because of how we're using it here

Reply:
Sold, they now all default to `List.Empty`, resolved on https://github.com/triaxtec/openapi-python-client/pull/426/commits/ff7950f8a767a5185f4893748b99d1f43cd89c48